File: TitikTengahDuaTitik2.java
import java.util.Scanner;

public class TitikTengahDuaTitik2 {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        System.out.print("Masukkan x untuk P1: ");
        int p1X = in.nextInt();

        System.out.print("Masukkan y untuk P1: ");
        int p1Y = in.nextInt();

        System.out.print("Masukkan x untuk P2: ");
        int p2X = in.nextInt();

        System.out.print("Masukkan y untuk P2: ");
        int p2Y = in.nextInt();

        Titik p1 = new Titik(p1X, p1Y);
        Titik p2 = new Titik(p2X, p2Y);
        Titik p3 = new Titik((p1.x + p2.x) / 2, (p1.y + p2.y) / 2);

        System.out.println();
        System.out.println("P3: (" + p3.x + ", " + p3.y + ")");

        in.close();
    }
}

File: titik_tengah_dua_titik2.py
class Titik:
  def __init__(self, x, y):
      self.x = x
      self.y = y
      
print('Masukkan x untuk P1: ', end='')
p1X = int(input())

print('Masukkan y untuk P1: ', end='')
p1Y = int(input())

print('Masukkan x untuk P2: ', end='')
p2X = int(input())

print('Masukkan y untuk P2: ', end='')
p2Y = int(input())

p1 = Titik(p1X, p1Y)
p2 = Titik(p2X, p2Y)
p3 = Titik((p1.x + p2.x) // 2, (p1.y + p2.y) // 2)

print()
print('P3: (' + str(p3.x) + ', ' + str(p3.y) + ')')